rename.cc

```diff
@@ -0,0 +1,54 @@
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include "./capsicum-test.h"
+#define RENAME_TEST_FILENAME        ".rename_test"
+#define RENAMEAT_TEST_FILENAME      ".renameat_test"
+#define RENAMEAT_TEST_DIR           ".renameat_testdir"
+
+/*
+added to test the renameat syscall for the case that
+    - the "to" file already exists
+    - the "to" file is specified by an absolute path
+    - the "to" file descriptor is used
+
+details at: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=222258
```

Comment:
Probably worth mentioning here that this was a Capsicum-related regression in base behaviour, which explains why this test doesn't include any Capsicum functionality.